waspc/cli/src/Wasp/Cli/Command/Requires.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Wasp.Cli.Command.Requires
+  ( -- * Using command requirements
+
+    -- This module exports the lower-level parts of the command requirement
+    -- API. To use these features to establish a requirement for a command,
+    -- use 'Wasp.Cli.Command.Common.commandRequires'. For convenience,
+    -- 'CommandRequirement' and its constructors are also exported from that
+    -- module.
+    --
+    -- Call @commandRequires SomeRequirementConstructor@ anywhere in your command,
+    -- preferably near the beginning of the command implementation and definitely
+    -- before using the requirement.
+    CommandRequirement (..),
+    RequiresT,
+    runRequiresT,
+    MonadRequires (..),
+  )
+where
+
+import Control.Monad.Error.Class (MonadError)
+import Control.Monad.IO.Class (MonadIO)
+import Control.Monad.State.Strict (StateT, evalStateT, gets, modify, when)
+import Control.Monad.Trans (MonadTrans)
+import Data.Foldable (find)
+
+-- | A requirement that a command needs in order to run
+data CommandRequirement
+  = -- | Requires that it is possible to connect to the prisma database.
+    DbConnection
+  deriving (Eq, Show)
+
+-- | A monad transformer that holds a list of satisfied 'CommandRequirement's
+-- that is updated whenever a new requirement is checked.
+--
+-- This is the canonical implementation of 'MonadRequires'.
+newtype RequiresT m a = RequiresT {unRequiresT :: StateT [CommandRequirement] m a}
+  deriving (Functor, Applicative, Monad, MonadTrans, MonadIO, MonadError b)
+
+runRequiresT :: Monad m => RequiresT m a -> m a
+runRequiresT m = evalStateT (unRequiresT m) []
+
+-- | The Requires monad keeps track of which 'CommandRequirements' have been
+-- checked and met.
+--
+-- NOTE: We go through the trouble of creating this abstraction so we can make
+-- sure the set of satisfied requirements is never modified in a way we do not
+-- want it to be (e.g. marking a requirement as met without actually checking
+-- it).
+class Monad m => MonadRequires m where
+  -- | @checkRequirement requirement check@ runs @check@ if the requirement
+  -- has not been met yet. @check@ should return whether the requirement is
+  -- met.
+  --
+  -- @checkRequirement requirement check@ returns @True@ when the requirement
+  -- is met, and will not run the check if the requirement has already been
+  -- checked and met.
```

Comment:
Seems like you duplicated the comment here, kind of?

I think it would be enough to just say:
```
@checkRequirement requirement doCheck@ returns True if specified requirement is met, otherwise False. It will run `doCheck` to check this, unless it already knows from before that requirement is met.
```

Reply:
What may be important to mention here is that this interface assumes that once a requirement is met, it stays met.